src/fheroes2/army/army_bar.cpp

```diff
@@ -585,6 +591,11 @@ bool ArmyBar::ActionBarLeftMouseDoubleClick( ArmyTroop & troop )
             }
         }
 
+        if ( can_change && keepLastTroop ) {
```

Comment:
The `Dialog::DISMISS` flag should be set if we do not need to keep the last troop in the army:
```suggestion
        if ( can_change && !keepLastTroop ) {
```